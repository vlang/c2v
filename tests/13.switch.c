#include <stdio.h>

const int a = 0;

void main() 
{ 
    switch(a)
    {
    default:
        printf("foo\n");
        break;
    case 0:
        printf("bar\n");
        break;
    }


    switch(a)
    {
    case 0:
        printf("bar\n");
        break;
    default:
        printf("foo\n");
        break;
    }
}