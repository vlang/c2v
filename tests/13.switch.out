[translated]
module main

[export: 'a']
const (
	a = 0
)

fn main() {
	match a {
		0 { // case comp body kind=CallExpr is_enum=true
			C.printf(c'bar\n')
		}
		else {
			C.printf(c'foo\n')
		}
	}
	match a {
		0 { // case comp body kind=CallExpr is_enum=true
			C.printf(c'bar\n')
		}
		else {
			C.printf(c'foo\n')
		}
	}
}
